### PR TITLE
mark github action working dir as safe for git

### PR DESCRIPTION
**Pull request by jeffatgametime**
[git 2.35.2 introduced the concept of safe directories](https://github.com/actions/runner-images/issues/6775#issuecomment-1352973077); so allow our action to run in the github working dir by declaring it safe.

[problem is](https://github.com/gametimesf/the-algo/actions/runs/4139814331/jobs/7157821902) that the generate command looks for the string `"nothing to commit, working tree clean"` in stdout from `git status` and fails the action if the exit code is not successful.

this worked when i pointed to this branch in [this pr](https://github.com/gametimesf/the-algo/pull/496)

### CODE REVIEW

**Comment by ajanata** [comment]
I would rather we spend some time trying to understand why it suddenly broke and how to fix it without disabling a security feature; I was planning on doing so this afternoon.

File: go/generate-clean/entrypoint.sh [hunk 1]
```diff
@@ -4,6 +4,8 @@ set -e
 git config --global url."https://${ORG_GITHUB_TOKEN}@github.com/gametimesf".insteadOf "https://github.com/gametimesf"
 export PATH="$PATH:/"
 
+git config --global --add safe.directory /github/workspace
```
Comment (ajanata) [inline go/generate-clean/entrypoint.sh:7]: can we try [this](https://github.com/espressif/github-actions/pull/35/files) instead?

Comment (jeffatgametime) [reply go/generate-clean/entrypoint.sh]: the change here worked, but the [referenced fix didn't](https://github.com/gametimesf/the-algo/actions/runs/4139543302/jobs/7208872071)

**Review by waitstime — APPROVED** [review]

File: go/generate-clean/entrypoint.sh (hunk 1 above)
Comment (ajanata) [reply go/generate-clean/entrypoint.sh:7]: annoying

**Review by ajanata — APPROVED** [review]
not a fan of this particular change but in the interest of unblocking shit

**Pull request merged**